massa-bootstrap/src/client_binder.rs

```diff
@@ -1,58 +1,54 @@
 // Copyright (c) 2022 MASSA LABS <info@massa.net>
 
+use std::time::Duration;
+
 use crate::error::BootstrapError;
 use crate::establisher::Duplex;
 use crate::messages::{
     BootstrapClientMessage, BootstrapClientMessageSerializer, BootstrapServerMessage,
     BootstrapServerMessageDeserializer,
 };
 use crate::settings::BootstrapClientConfig;
-use async_speed_limit::clock::StandardClock;
-use async_speed_limit::{Limiter, Resource};
 use massa_hash::{Hash, HASH_SIZE_BYTES};
 use massa_models::serialization::{DeserializeMinBEInt, SerializeMinBEInt};
 use massa_models::version::{Version, VersionSerializer};
 use massa_serialization::{DeserializeError, Deserializer, Serializer};
 use massa_signature::{PublicKey, Signature, SIGNATURE_SIZE_BYTES};
 use rand::{rngs::StdRng, RngCore, SeedableRng};
-use tokio::io::AsyncReadExt;
-use tokio::io::AsyncWriteExt;
 
 /// Bootstrap client binder
-pub struct BootstrapClientBinder {
+pub struct BootstrapClientBinder<D: Duplex> {
     // max_bootstrap_message_size: u32,
     size_field_len: usize,
     remote_pubkey: PublicKey,
-    duplex: Resource<Duplex, StandardClock>,
+    duplex: D,
     prev_message: Option<Hash>,
     version_serializer: VersionSerializer,
     cfg: BootstrapClientConfig,
 }
 
-impl BootstrapClientBinder {
+impl<D: Duplex> BootstrapClientBinder<D> {
     /// Creates a new `WriteBinder`.
     ///
     /// # Argument
     /// * duplex: duplex stream.
     /// * limit: limit max bytes per second (up and down)
     #[allow(clippy::too_many_arguments)]
-    pub fn new(duplex: Duplex, remote_pubkey: PublicKey, cfg: BootstrapClientConfig) -> Self {
+    pub fn new(duplex: D, remote_pubkey: PublicKey, cfg: BootstrapClientConfig) -> Self {
         let size_field_len = u32::be_bytes_min_length(cfg.max_bootstrap_message_size);
         BootstrapClientBinder {
             size_field_len,
             remote_pubkey,
-            duplex: <Limiter>::new(cfg.max_bytes_read_write).limit(duplex),
+            duplex,
```

Comment:
This will need to be reinstated at some point using the library written by Aurelien

Reply:
added to #3746